CMakeLists.txt

```
@@ -1,5 +1,28 @@
 cmake_minimum_required(VERSION 3.5..3.31)
 
+if(USE_CPP_INJECT_HASH)
+    # Configure LIEF build options first
+    set(LIEF_PYTHON_API OFF CACHE BOOL "")
+    set(LIEF_EXAMPLES OFF CACHE BOOL "")
+    set(LIEF_TESTS OFF CACHE BOOL "")
+    set(LIEF_DOC OFF CACHE BOOL "")
+    set(LIEF_INSTALL ON CACHE BOOL "")
+    set(LIEF_STATIC ON CACHE BOOL "")
+    set(LIEF_ELF ON CACHE BOOL "Enable ELF format")
+    set(LIEF_PE ON CACHE BOOL "Enable PE format")
+    set(LIEF_MACHO ON CACHE BOOL "Enable MACHO format")
+    set(LIEF_DEX OFF CACHE BOOL "Disable DEX format")
+    set(LIEF_VDEX OFF CACHE BOOL "Disable VDEX format")
+    set(LIEF_ART OFF CACHE BOOL "Disable ART format")
+    set(LIEF_OAT OFF CACHE BOOL "Disable OAT format")
+    
+    # Add LIEF subdirectory
+    add_subdirectory(third_party/lief)
```

It is not necessary to comment every line, especially when they are self explanatory like this. 
```suggestion
    add_subdirectory(third_party/lief)
```